packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -1061,11 +1061,15 @@ const VirtualListBaseFactory = (type) => {
 		getScrollWidth = () => (this.isPrimaryDirectionVertical ? this.scrollBounds.clientWidth : this.getVirtualScrollDimension())
 
 		getVirtualScrollDimension = () => {
-			const
-				{dimensionToExtent, primary, curDataSize} = this,
-				{spacing} = this.props;
+			if (this.props.itemSizes) {
+				return this.props.itemSizes.reduce((total, size) => total + size);
```

Comment:
We assumed that an app gives an array of item size, so it looks simple and good way to calculate the total size. Just one case I concern is that there is actually no item at all. In this case, `itemSizes` could be an empty array and `TypeError` will occur in this line. So it looks better to specify the initial value.

Reply:
@0x64 
For an empty array, I added the initial value of 0.
For null or undefined, `this.props.itemSizes` is false.
For other types, `itemSizes` was defined as `PropTypes.arrayOf(PropTypes.number)`. So other types can't be acceptable.